exporter/util.go

```diff
@@ -142,12 +148,24 @@ func (ic *importContext) emitNotebookOrRepo(path string) {
 
 func (ic *importContext) getAllDirectories() []workspace.ObjectStatus {
 	if len(ic.allDirectories) == 0 {
-		notebooksAPI := workspace.NewNotebooksAPI(ic.Context, ic.Client)
-		ic.allDirectories, _ = notebooksAPI.ListDirectories("/", true, true)
+		objects := ic.getAllWorkspaceObjects()
+		for _, v := range objects {
+			if v.ObjectType == workspace.Directory {
+				ic.allDirectories = append(ic.allDirectories, v)
+			}
+		}
 	}
 	return ic.allDirectories
 }
 
+func (ic *importContext) getAllWorkspaceObjects() []workspace.ObjectStatus {
+	if len(ic.allWorkspaceObjects) == 0 {
```

Comment:
I suggest we use a mutex here to protect access to this field

Reply:
Right now exporter is single threaded, so no need for the mutex. I'm thinking on how to make it running in parallel, but it requires more thinking.

Reply:
Gotcha. At least this is in one method. We'll address multithreading later then.